projects/kit/components/tree/components/tree-item-content/tree-item-content.component.ts

```diff
@@ -20,6 +20,7 @@ import {TUI_TREE_CONTROLLER} from '../../misc/tree.tokens';
     changeDetection: ChangeDetectionStrategy.OnPush,
     host: {
         '[class._expandable]': 'isExpandable',
+        '(click)': 'onClick()',
```

Comment:
Moving it here means now whatever is in the content, be it a checkbox or any other interactive element - whenever you click it, it would toggle the tree item. We shouldn't move click handler here from a button.